draft-enghardt-panrg-path-properties.md

```diff
@@ -96,19 +96,19 @@ Link:
 
 Path:
-: A sequence of directly adjacent path elements, alternating between nodes and links. A path starts and ends with a host and can be traversed by packets. A path is defined on a specific layer. For example, a path on the network layer is a sequence of path elements between two network layer nodes that can handle packets addressed from or to themselves on the network layer. With multicast or broadcast, a single packet may be sent over multiple paths at once -- one path for each combination of sender and receiver.
+: A sequence of directly adjacent path elements, alternating between nodes and links, starting and ending with a host. A path is defined on a specific layer. For example, a path on the network layer is a sequence of adjacent path elements implementing network layer functionality between two hosts. Paths can be traversed by packets. With multicast or broadcast, a single packet may be sent over multiple paths at once -- one path for each combination of sending and receiving host.
 
 Subpath:
 : Given a path, a subpath is a sequence of directly adjacent path elements of this path, starting and ending with a node.
 
 Flow:
-: One or multiple packets which are traversing the same subpath or path. For example, a flow can consist of all packets sent within a TCP session with the same five-tuple between two hosts.
+: One or multiple packets which are traversing the same subpath or path. For example, a flow can consist of all packets sent within a TCP session with the same five-tuple between two hosts, or it can consist of all packets sent on the same physical link.
 
 Property:
-: A trait of one or multiple path elements of a path, or a trait of a flow with respect to one or multiple path elements. An example of a property of a link is the maximum data rate that can be sent over this link. An example of a property of a node is the administrative domain that the node belongs to. An example of a property of a flow with respect to one or multiple path elements is the one-way delay of a packet being sent from one node to another node over a subpath.
+: A trait of one or multiple path elements, or a trait of a flow with respect to one or multiple path elements. An example of a link property is the maximum data rate that can be sent over the link. An example of a node property is the administrative domain that the node belongs to. An example of a property of a flow with respect to a subpath is the aggregated one-way delay of the flow being sent from one node to another node over a subpath.
 A property is thus described by a tuple containing the set of path elements, the flow or an empty set if no packets are relevant for the property, the name of the property (e.g., maximum data rate), and the value of the property (e.g., 1Gbps).
```

Comment:
Maybe we should define a property on a sequence of path elements, since if we talk about one way latency between two nodes A and B, the order of the nodes changes the meaning of the property.

Reply:
Agreed.
I wanted to be able to define a property of, e.g., just some nodes but not the links between them, but I'm not sure if we actually need this, and if we do, I think we can work with it anyway.

Reply:
We can also define it as: "... containing the sequence or set of path elements", but this makes the concept more complex which is probably not a good idea.

Since we define it as a "sequence of path elements" and not "sequence of path elements alternating between links and nodes", we can still define a property for some nodes (they would just be ordered).

Reply:
Phew, tricky. :D Yes, I think sequence is fine. If we must have the set, we can have it as a set of single properties, each of which is defined for a specific path element.